File: Linked-List-Algorithms/Algorithms in C#/midSinglyLinkedList.cs
public class Program
{
    // create a Linked List node
    public class Node
    {
        public int data;
        public Node next;

        public Node(int data)
        {
            this.data = data;
            this.next = null;
        }
    }

    public static Node head;

    public static void Main(string[] args)
    {
        // create a linked list with 5 elements
        head = new Node(1);
        head.next = new Node(2);
        head.next.next = new Node(3);
        head.next.next.next = new Node(4);
        head.next.next.next.next = new Node(5);

        int middleElement = FindMiddleElement(head);

        Console.WriteLine("Middle element of the linked list is " + middleElement);
    }

    // Function to find the middle element of a singly linked list
    private static int FindMiddleElement(Node head)
    {
        Node slow_ptr = head;
        Node fast_ptr = head;

        if (head != null)
        {
            while (fast_ptr != null && fast_ptr.next != null)
            {
                fast_ptr = fast_ptr.next.next;
                slow_ptr = slow_ptr.next;
            }
            return slow_ptr.data;
        }
        return 0;
    }
}

File: Linked-List-Algorithms/Algorithms in C++/midSinglyLinkedList.cpp
#include <iostream>
using namespace std;

// create a node structure 
struct Node 
{ 
    int data; 
    Node* next; 
}; 
  
// function to get the middle of the linked list 
void printMiddle(Node *head) 
{ 
    Node *slow_ptr = head; 
    Node *fast_ptr = head; 
  
    if (head!=NULL) 
    { 
        while (fast_ptr != NULL && fast_ptr->next != NULL) 
        { 
            fast_ptr = fast_ptr->next->next; 
            slow_ptr = slow_ptr->next; 
        } 
        cout<<"The middle element is "<<slow_ptr->data; 
    } 
} 
  
// function to insert a new node at the beginning 
void push(Node** head_ref, int new_data) 
{ 
    Node* new_node = new Node(); 
  
    new_node->data  = new_data; 
  
    new_node->next = (*head_ref); 
  
    (*head_ref)    = new_node; 
} 
  
// function to print the linked list 
void printList(Node *head) 
{ 
    while (head != NULL) 
    { 
        cout << head->data << " "; 
        head = head->next; 
    } 
    cout<<"\n"; 
} 
  
// Driver code 
int main() 
{ 
    Node* head = NULL; 
    int i; 
  
    for (i=5; i>0; i--) 
    { 
        push(&head, i); 
        printList(head); 
        printMiddle(head); 
    } 
  
    return 0; 
}